ports/espressif/common-hal/sdioio/SDCard.c

```diff
@@ -0,0 +1,223 @@
+// This file is part of the CircuitPython project: https://circuitpython.org
+//
+// SPDX-FileCopyrightText: Copyright (c) 2024 Jacob Rigby
+//
+// SPDX-License-Identifier: MIT
+
+#include <stdbool.h>
+#include "esp_err.h"
+
+#include "driver/sdmmc_host.h"
+#include "ports/espressif/esp-idf/components/sdmmc/include/sdmmc_cmd.h"
+
+#include "shared-bindings/microcontroller/Pin.h"
+#include "shared-bindings/util.h"
+#include "shared-bindings/sdioio/SDCard.h"
+#include "py/mperrno.h"
+#include "py/runtime.h"
+
+#include "esp_log.h"
+static const char *TAG = "SDCard.c";
+
+static bool sdio_host_init = false;
+
+static void common_hal_sdioio_sdcard_check_for_deinit(sdioio_sdcard_obj_t *self) {
+    if (common_hal_sdioio_sdcard_deinited(self)) {
+        raise_deinited_error();
+    }
+}
+
+static int check_pins(const mcu_pin_obj_t *clock, const mcu_pin_obj_t *command, const uint8_t num_data, const mcu_pin_obj_t **data) {
+    if (CONFIG_SOC_SDMMC_USE_GPIO_MATRIX) {
```

Comment:
```suggestion
    #ifdef CONFIG_SOC_SDMMC_USE_GPIO_MATRIX
```

Use a preprocessor `#if` to test against a preprocessor macro. (clang complains if you don't.)

Reply:
Thanks, done.